### ISSUE TITLE
Parse floats using fast path?

### ISSUE THREAD

**Issue by kangalioo** [Alexhuszagh/rust-lexical#40]
## Problem
In my application, over 25% of execution time is spent inside `lexical::parse_lossy`. Mind you, lexical's implementation is _far_ better than the stdlib implementation and its speed is simply fantastic.

A little bit more speed can't hurt though, so I was looking at the [implementation details](https://github.com/Alexhuszagh/rust-lexical/tree/master/lexical-core#string-to-float) where is stated that `parse_lossy` tries multiple parsing implementations: first the fast path, then the moderate path. Is it possible to make lexical return the fast path's result directly?

To give context; this is an excerpt of the floats that need to be parsed:

```rust
-0.018477, -0.018464, -0.018458, -0.014031, -0.014018, -0.014011, -0.000648, 0.008092, 
0.000111, -0.009875, 0.012704, 0.012185, 0.011334, 0.011927, 0.012284, 0.010097,
0.012951, 0.001517, -0.005452, 0.015123, -0.004884, -0.007977, 0.019697, 0.010684
```
They're all in between -1.0 and +1.0, and only the first few 4-5 floating point digits matter.

## Solution
It could be implemented using a new parse function, maybe `lexical::parse_lossier`?

**Comment by myrrlyn** [comment]
I'll take a look at doing this either next month or August. Thank you for your request and your patience

**Comment by Alexhuszagh** [comment]
@kangalioo That should be doable, definitely. I could just expose the fast path algorithm directly. I'll add this for a feature for the next minor release.

**Comment by Alexhuszagh** [comment]
This has been added as of recent commits for the [v0.8](https://github.com/Alexhuszagh/rust-lexical-experimental) branch. I'll post a working sample once this full functionality has been released.

**Comment by Alexhuszagh** [comment]
This has been closed as of the latest commit, I'll keep this open until I publish v0.8, however.

This isn't extensively documented, since it won't be a correct implementation, however, this will work using the following code:

```rust
use lexical_parse_float::Options;
use lexical_parse_float::format::STANDARD;
use lexical_parse_float::parse::ParseFloat;

let options = Options::new();
f64::fast_path_complete::<{ STANDARD }>(b"1.34000", &options);
```

I'll add a line or two to the documentation about this, but I'm not planning on making the API much nicer, but I will guarantee stability for this.

**Comment by Alexhuszagh** [comment]
Implemented as of [lexical v6.0.0](https://crates.io/crates/lexical/6.0.0) and [lexical-core v0.8.0](https://crates.io/crates/lexical-core/0.8.0). Thanks for the patience ^_^.
